spec/integration/data_stream/routing_rules.spec.yml

```
@@ -0,0 +1,74 @@
+##
+## Describes the specification for a routing rules yml file
+##
+spec:
+  # Everything under here follows JSON schema (https://json-schema.org/), written as YAML for readability
+  definitions:
+    routing_rule:
+      description: Routing rule definition (technical preview)
+      type: object
+      properties:
+        target_dataset:
+          description: >
+            Field references or a static value for the dataset part of the data stream name.
+            In addition to the criteria for index names, cannot contain - and must be no longer than 100 characters.
+            Example values are nginx.access and nginx.error.
+
+            Supports field references with a mustache-like syntax (denoted as {{double}} or {{{triple}}} curly braces).
+            When resolving field references, the processor replaces invalid characters with _.
+            Uses the <dataset> part of the index name as a fallback if all field references resolve to a null, missing, or non-string value
+          anyOf:
+            - type: string
+            - type: array
+              items:
+                type: string
+          examples:
+          - nginx.error
+          - nginx
+        if:
+          description: Conditionally execute the processor
+          type: string
+          examples:
+          - "ctx?.file?.path?.contains('/var/log/nginx/error')"
+          - "ctx?.container?.image?.name == 'nginx'"
+        namespace:
+          description: >
+            Field references or a static value for the namespace part of the data stream name.
+            See the criteria for index names for allowed characters. Must be no longer than 100 characters.
+
+            Supports field references with a mustache-like syntax (denoted as {{double}} or {{{triple}}} curly braces).
+            When resolving field references, the processor replaces invalid characters with _.
+            Uses the <namespace> part of the index name as a fallback if all field references resolve to a null, missing, or non-string value.
+          anyOf:
+            - type: string
+            - type: array
+              items:
+                type: string
+          items:
+            type: string
+          examples:
+            - default
+            - "{{ labels.dasta_stream.namespace}}"
+      required:
+        - target_dataset
+        - if
+        - namespace
+  type: array
+  description: Routing rules set.
+  items:
+     type: object
+     additionalProperties: false
+     properties:
+       source_dataset:
+         description: >
+           Source dataset to be used by this reroute processsor.
+           If applicable, documents from this dataset will be routed according to the rules defined.
+         type: string
+       rules:
+         description: List of routing rules
+         type: array
+         items:
+           $ref: "#/definitions/routing_rule"
```

Nit. Should we require this to have at least one rule?
```suggestion
         type: array
         minItems: 1
         items:
           $ref: "#/definitions/routing_rule"
```

---

Yes, that would be better to avoid empty files or empty arrays.

---

There is an error that could be related to the conversion it is done between YAML and JSON

> 1. could not read root folder spec file: could not load spec for "integration/data_stream/spec.yml": could not load schema for "integration/data_stream": failed to load schema for "integration/data_stream/routing_rules.spec.yml": minItems must be of an integer


I'll remove this `minItems`